lib/account.php

```diff
@@ -292,11 +292,19 @@ public function deleteMessage($sourceFolderId, $messageId) {
 		$hordeSourceMailBox = new Horde_Imap_Client_Mailbox($sourceFolderId);
 		$hordeTrashMailBox = new Horde_Imap_Client_Mailbox($trashId);
 
-		$result = $this->getImapConnection()->copy($hordeSourceMailBox, $hordeTrashMailBox,
-			array('create' => $createTrash, 'move' => true, 'ids' => $hordeMessageIds));
+		if ($hordeSourceMailBox == $hordeTrashMailBox) {
```

Comment:
This check could be potentially dangerous, if the values inside the objects get out of sync. Perhaps check if the IDs are equal, since they are strings? Also, you can then use the safer `===` operator.